### PR TITLE
[FEATURE] FIAMS CV params in mzML

### PR DESCRIPTION
**Pull request by jpfeuffer**
# Description

Adds new mzML CV params. See gitter.

# Checklist:
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes

### CODE REVIEW

**Comment by timosachsenberg** [comment]
also make sure it is written out in the same way... otherwise, it gets lost

**Comment by jpfeuffer** [comment]
Question @OpenMS/core : Would it make sense to parse all CV params by default?

**Comment by jpfeuffer** [comment]
@timosachsenberg We do not have a writing mechanism for parsed CV terms. So this is currently not possible.

**Comment by timosachsenberg** [comment]
I think this would make sense but sounds quite involved. Could potentially also consume more memory.

**Comment by hroest** [comment]
what about writing it back again? with the right term?

**Comment by hroest** [comment]
> @timosachsenberg We do not have a writing mechanism for parsed CV terms. So this is currently not possible.
> I.e. we would need an equally long if-chain to write the values out.

I think that is what we currently do, isnt it?

**Comment by jpfeuffer** [comment]
Ah maybe you are right. Does it try to look up the name (not ID) of the CV term here?
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp#L3468

I thought we would need to store the ID as well.

**Comment by timosachsenberg** [comment]
ah ok I meant you want to store it as cvTerm on the MSExperiment side, too.

**Comment by jpfeuffer** [comment]
Do we need to update the PSI.obo for this?

**Comment by jpfeuffer** [comment]
Actually after getting the file, it seems like this is the only term missing in our parser.

**Comment by jpfeuffer** [comment]
Unfortunately, the CV term is registered completely wrong. It is registered as a *scan* not *spectrum* attribute and does not even have an xref for the data type specified.

**Comment by jpfeuffer** [comment]
If it is instead really a scan attribute, then the whatever converter was used writes it wrong.

**Comment by jpfeuffer** [comment]
@hroest Where does this attribute fit better in your opinion? In the spectrum or spectrum/scanList/scan element? Then we can raise issues upstream.

**Comment by jpfeuffer** [comment]
By the way, there is already an entry (filter string) in the spectrum/scanList/scan, that contains the "**c**ompensation **v**oltage".

```
<cvParam cvRef="MS" accession="MS:1000512" name="filter string" value="FTMS + p NSI cv=-50.00 Full ms [197.0777-1500.0000]"/>
              <cvParam cvRef="MS" accession="MS:1000927" name="ion injection time" value="50.0" unitCvRef="UO" unitAccession="UO:0000028" unitName="millisecond"
```

**Comment by jpfeuffer** [comment]
So it seems like it indeed should be placed and expected in the scan element but it was not yet fixed upstream in pwiz.
https://github.com/ProteoWizard/pwiz/issues/747

When I find time I will update this PR to do it the correct way. Seems like pwiz plans to fix it in a non-backward compatible way anyhow (which makes sense).

File: src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp [hunk 1]
```diff
@@ -1644,6 +1644,14 @@ namespace OpenMS
           //No member => meta data
           spec_.setMetaValue("peak list raw scans", termValue);
         }
+        else if (accession == "MS:1001581") //FAIMS compensation voltage
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp:1647]: below "spectrum" parent

File: src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp [hunk 2]
```diff
@@ -1741,7 +1749,7 @@ namespace OpenMS
             chromatogram_.getPrecursor().getPossibleChargeStates().push_back(value.toInt());
           }
         }
-        else if (accession == "MS:1002476" || accession == "MS:1002815") //ion mobility drift time
+        else if (accession == "MS:1002476" || accession == "MS:1002815" || accession == "MS:1001581") //ion mobility drift time or FAIM compensation voltage
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp:1752]: below "scan"

File: src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp [hunk 3]
```diff
@@ -2081,7 +2093,7 @@ namespace OpenMS
           //No member => meta data
           spec_.setMetaValue("dwell time", termValue);
         }
-        else if (accession == "MS:1002476" || accession == "MS:1002815") //ion mobility drift time
+        else if (accession == "MS:1002476" || accession == "MS:1002815" || accession == "MS:1001581") //ion mobility drift time or FAIMS compensation voltage
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp:2096]: below precursor (probably not used)

File: src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp [hunk 4]
```diff
@@ -5121,8 +5139,13 @@ namespace OpenMS
         {
           os << "\t\t\t\t\t\t<cvParam cvRef=\"MS\" accession=\"MS:1000016\" name=\"scan start time\" value=\"" << spec.getRT()
              << "\" unitAccession=\"UO:0000010\" unitName=\"second\" unitCvRef=\"UO\" />\n";
-          // if drift time was never set, don't report it
-          if (spec.getDriftTime() >= 0.0)
+
+          if (spec.getDriftTimeUnit() == MSSpectrum::DriftTimeUnit::FAIMS_COMPENSATION_VOLTAGE)
+          {
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp:5144]: writing out below "scan" only

**Comment by jpfeuffer** [comment]
Great! Thanks @timosachsenberg

**Pull request merged**